python/lsst/analysis/tools/actions/vector/calcRhoStatistics.py

```diff
@@ -0,0 +1,649 @@
+            ),
+            "Linear": (
+                "Linear binning in the distance. The bin steps will be "
+                "uniform in r from min_sep .. max_sep."
+            ),
+            "TwoD": (
+                "2-dimensional binning from x = (-max_sep .. max_sep) "
+                "and y = (-max_sep .. max_sep). The bin steps will be "
+                "uniform in both x and y. (i.e. linear in x,y)"
+            ),
+        },
+    )
+
+    min_rpar = Field[float](
+        doc=(
+            "The minimum difference in Rparallel to allow for pairs being "
+            "included in the correlation function. "
+        ),
+        default=None,
+        optional=True,
+    )
+
+    max_rpar = Field[float](
+        doc=(
+            "The maximum difference in Rparallel to allow for pairs being "
+            "included in the correlation function. "
+        ),
+        default=None,
+        optional=True,
+    )
+
+    period = Field[float](
+        doc="For the 'Periodic' metric, the period to use in all directions.",
+        default=None,
+        optional=True,
+    )
+
+    xperiod = Field[float](
+        doc="For the 'Periodic' metric, the period to use in the x direction.",
+        default=None,
+        optional=True,
+    )
+
+    yperiod = Field[float](
+        doc="For the 'Periodic' metric, the period to use in the y direction.",
+        default=None,
+        optional=True,
+    )
+
+    zperiod = Field[float](
+        doc="For the 'Periodic' metric, the period to use in the z direction.",
+        default=None,
+        optional=True,
+    )
+
+    var_method = ChoiceField[str](
+        doc="Which method to use for estimating the variance",
+        default="shot",
+        optional=True,
+        allowed={
+            method: method
+            for method in [
+                "shot",
+                "jackknife",
+                "sample",
+                "bootstrap",
+                "marked_bootstrap",
+            ]
+        },
+    )
+
+    num_bootstrap = Field[int](
+        doc=("How many bootstrap samples to use for the 'bootstrap' and " "'marked_bootstrap' var methods."),
+        default=500,
+        optional=True,
+    )
+
+    num_threads = Field[int](
+        doc=(
+            "How many OpenMP threads to use during the calculation. "
+            "If set to None, use all the available CPU cores."
+        ),
+        default=None,
+        optional=True,
+    )
+
+    def validate(self):
+        # Docs inherited from base class
+        super().validate()
+        req_params = (self.nbins, self.bin_size, self.min_sep, self.max_sep)
+        num_req_params = sum(param is not None for param in req_params)
+        if num_req_params != 3:
+            msg = (
+                "You must specify exactly three of ``nbins``, ``bin_size``, ``min_sep`` and ``max_sep``"
+                f" in treecorr_config. {num_req_params} parameters were set instead."
+            )
```

Comment:
I guess this answered my question, would it be useful though to have defaults set for the parameters?

Reply:
I don't think a reasonable defaults can be set that is agnostic to the data/problem at hand. I believe that is why `treecorr` does not have defaults for these either. The default values for all other fields correspond to the default values of the corresponding quantities in `treecorr`. So for the sake of consistency, I'd like to leave them without any defaults.